Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -15,7 +16,176 @@ public extension PBXProj.Objects {
+        )
+        let reference = generateReference(newGroup, groupName)
+        addObject(newGroup, reference: reference)
+        parentGroup.children.append(reference)
+        return (reference, newGroup)
+    }
+
+    /// Adds file at the give path to the project or returns existing file and its reference.
+    ///
+    /// - Parameters:
+    ///   - filePath: path to the file
+    ///   - sourceTree: file source tree, default is `.group`.
+    ///   - name: file name, by default gets file name from the path
+    /// - Returns: new or existing file and its reference.
+    public func addFile(at filePath: Path, sourceRoot: Path) throws -> (String, PBXFileReference) {
+        guard filePath.isFile else { throw XCodeProjEditingError.notAFile(path: filePath) }
+
+        if let existingFileReference = fileReferences.first(where: {
+            filePath == fullPath(fileElement: $0.value, reference: $0.key, sourceRoot: sourceRoot)
+        }) {
+            return existingFileReference
+        }
+
+        let fileReference = PBXFileReference(
+            sourceTree: .group,
+            name: filePath.lastComponent,
+            explicitFileType: PBXFileReference.fileType(path: filePath),
+            lastKnownFileType: PBXFileReference.fileType(path: filePath),
+            path: filePath.string
+        )
+        let reference = generateReference(fileReference, filePath.string)
+        addObject(fileReference, reference: reference)
+        return (reference, fileReference)
+    }
+
+    /// Adds file to the given group.
+    /// If group already contains file with given reference method does nothing.
+    ///
+    /// - Parameter group: group to add file to
+    /// - Parameter reference: file reference
+    public func addFile(toGroup group: PBXGroup, reference: String) {
+        guard !group.children.contains(reference) else { return }
+        group.children.append(reference)
+    }
+
+    /// Adds file to the given target's sources build phase or returns existing build file and its reference.
+    /// If target's sources build phase can't be found returns nil.
+    ///
+    /// - Parameter target: target obejct
+    /// - Parameter reference: file reference
+    /// - Returns: new or existing build file and its reference
+    public func addBuildFile(toTarget target: PBXTarget, reference: String) -> (String, PBXBuildFile)? {
+        guard let sourcesBuildPhase = sourcesBuildPhase(target: target) else { return nil }
+        if let existingBuildFile = buildFiles.first(where: { $0.value.fileRef == reference }) {
+            return existingBuildFile
+        }
+
+        let buildFile = PBXBuildFile(fileRef: reference)
+        let reference = generateReference(buildFile, reference)
+        addObject(buildFile, reference: reference)
+        sourcesBuildPhase.files.append(reference)
+        return (reference, buildFile)
+    }
+
+    public func fullPath(fileElement: PBXFileElement, reference: String, sourceRoot: Path) -> Path? {
```

Comment:
could you add few unit tests to this function. I am not sure what cases it has to cover and how. Especially with groups.